shell/creators/extension/init

```diff
@@ -0,0 +1,139 @@
+args.forEach((arg, index) => {
+  switch (arg) {
+  case '--update':
+  case '-u':
+    updateOnly = true;
+    break;
+  case '--app-name':
+  case '-a':
+    if ( args[index + 1] && !args[index + 1].startsWith('-') ) {
+      appName = args[index + 1];
+    } else {
+      console.error('Error: Missing value for --app-name or -a option.');
+      process.exit(1);
+    }
+    break;
+  case '--skeleton-only':
+  case '-s':
+    skeletonOnly = true;
+    break;
+  default:
+    if ( !arg.startsWith('-') && extensionName === '' ) {
+      extensionName = arg;
+      appName = appName || extensionName;
+    }
+  }
+});
+
+if ( !extensionName && !updateOnly && !skeletonOnly ) {
+  console.error('Please provide an extension name.');
+  process.exit(1);
+}
+
+if ( !appName ) {
+  console.error('Error: The application name is not defined.');
+  process.exit(1);
+}
+
+const basePath = process.cwd();
+let skeletonPath;
+let isInsideSkeleton = false;
+let directoryExists = false;
+
+// Check if we are inside a skeleton application by looking for package.json
+if ( fs.existsSync(path.join(basePath, 'package.json')) ) {
+  isInsideSkeleton = true;
+  skeletonPath = basePath;
+} else {
+  // If not inside a skeleton, check if a directory with the appName already exists
+  skeletonPath = path.join(basePath, appName);
+
+  if ( fs.existsSync(skeletonPath) ) {
+    directoryExists = true;
+  }
+}
+
+const pkgPath = path.join(skeletonPath, 'pkg', extensionName);
+const updatePath = path.join(__dirname, 'update');
+
+const skeletonExists = fs.existsSync(skeletonPath);
+const pkgExists = fs.existsSync(pkgPath);
+
+try {
+  if ( updateOnly ) {
+    // Run the update script directly
+    console.log('Updating applications...');
+    execSync(`node ${ path.join(updatePath, 'init') }`, { stdio: 'inherit' });
+
+    console.log('Update completed successfully.');
+    process.exit(0);
+  }
+
+  // If the directory exists but we're not inside a skeleton, we should exit to prevent overwriting
+  if ( directoryExists && !isInsideSkeleton ) {
+    throw new Error(`A directory named "${ appName }" already exists. Aborting.`);
+  }
+
+  // Create skeleton application if it doesn't exist
+  if ( !isInsideSkeleton && !skeletonExists ) {
+    console.log(`Creating skeleton application: ${ appName }...`);
+    // Pass all arguments to the app/init script
+    execSync(`node ${ path.join(__dirname, 'app', 'init') } ${ appName } ${ args.join(' ') }`, { stdio: 'inherit' });
+
+    // Ensure the skeleton path directory is created before attempting to change directory
+    if ( !fs.existsSync(skeletonPath) ) {
+      throw new Error(`Failed to create skeleton application directory: ${ skeletonPath }`);
+    }
+
+    // Change working directory to the newly created skeleton app
+    process.chdir(skeletonPath);
+
+    // Install dependencies after creating the skeleton
+    console.log(`Installing dependencies in ${ appName }...`);
+    execSync(`yarn install`, { stdio: 'inherit' });
```

Comment:
This is because of the copying of the annotations, right?

Reply:
Yes, also for determining the latest shell package version. Perhaps we could think of a different way to preform these things? I would be nice to not enforce the installation of the modules first.

Reply:
Ah, that was quite a discussion me and @nwmac had back in the day. We couldn't find a better way of doing this... It's tied to the way we read the shell version from the dependencies. We have another go at it, but the enforcement has been there for a while, since we've been preparing the release of Rancher 2.9

Reply:
Would it be possible to make a call out to NPM or even Github to get the latest `@rancher/shell` version rather than using the node_modules?

Reply:
I've updated this with bc11f9dff6e707787d5e18ef0be10d19add1994b to now use `npm view @rancher/shell version` to retrieve the latest version from npm rather than using the installed node_modules.

So now the modules will not be installed during the creation, and following the creation of a skeleton app the user will receive a message:

```console
Extension created successfully.
To begin, run: 

        cd base42 && yarn install
```

> Note: the base42 was the name of my skeleton app